### PR TITLE
4818-client-name-permission-data-migration

### PR DESCRIPTION
**Pull request by ttoomey**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description
[GH Issue](https://github.com/open-path/Green-River/issues/4818)

Migration to set the can_view_client_name to true on existing roles that might be used to grant access to a client

## Type of change
[//]: # 'remove options that are not relevant'
- [x] Bug fix

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: db/migrate/20240601143623_client_name_permission_data_migration.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    view_cond = <<~SQL
+      can_view_clients = true OR
+      can_edit_clients = true OR
+      can_search_clients_with_roi = true OR
+      can_search_window = true OR
+      can_search_own_clients = true OR
+      can_search_all_clients = true
+    SQL
+    Role.where(view_cond).
+      where.not('name ILIKE ?', '%Green River%').
```
Comment (ttoomey) [inline db/migrate/20240601143623_client_name_permission_data_migration.rb:16]: intention is keep this off for GR staff

File: app/models/role.rb [hunk 2]
```diff
@@ -149,6 +149,7 @@ def self.administrative? permission:
     permissions_with_descriptions.merge(health_permissions_with_descriptions)[permission][:administrative] rescue true # rubocop:disable Style/RescueModifier
   end
 
+  # appears to be broken, the :access property is not present in permissions_with_descriptions
   def self.permissions_for_access(access)
     permissions_with_descriptions.select { |_k, attrs| attrs[:access].include?(access) }.keys
   end
```
Comment (ttoomey) [inline app/models/role.rb]: Noticed this issue while investigating:
This is called by `with_viewable_permissions` but throws an exception, maybe it's rotted away and should be removed?

Comment (eanders) [reply app/models/role.rb]: Thanks, looks like there are corollary versions in the HMIS side that _are_ used.  Removing this and a couple other unused methods that called it.

**Review by eanders — APPROVED** [review]
Thanks @ttoomey !  Tested locally with and without ACLs and at least on the surface it appears to work correctly.

**Pull request merged**